### PR TITLE
Support - LLM - Add a performance overlay to track navigation time between pages

### PR DESCRIPTION
**Pull request by nparigi-ledger**
### 📝 Description

Add a performance overlay to LLM

It displays the time that took to navigate to the current page, in the corner of the screen.
The time is calculated, using timing hooks provided by [react-native-performance](https://shopify.github.io/react-native-performance/docs/fundamentals/getting-started), from the moment that a press on a navigate button is handled by React-native, to the moment when destination screen is rendered and interactive.
The status color of the render report is based on [Coinbase React Native Vitals](https://www.coinbase.com/blog/performance-vitals-a-unified-scoring-system-to-guide-performance-health-and-prioritization)
The overlay UI is 100% borrowed from what Olivier have done with the analytics overlay :ninja:
I still need to tweak a little the timings settings according to the tests I have to do soon on the production, and i'm still not sure if i need to take into account the transition from react-navigation
Also, the timing is not 100% accurate according to an user point, as if the the JS thread is bloated, the button press timing is taken into account only when the JS thread is available again, but i don't think i can do anything about that ...

For now, the only tracked flow are : 
Portfolio page -> Asset -> Account
Portfolio page -> Assets

In production, the performance profiler is completely disabled.
Even if there are calls to the profiler in the code that will always be called even in production, they do nothing, and there is almost no additional logic, so insignificant impact on performance.

### ❓ Context

- **Impacted projects**: `` live-mobile
- **Linked resource(s)**: `` 

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

https://user-images.githubusercontent.com/89014981/216625275-2989aa01-9786-40ca-9b41-eafbf68c3165.mp4

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-mobile/src/screens/WalletCentricAsset/index.tsx [hunk 1]
```diff
@@ -297,34 +298,36 @@ const AssetScreen = ({ route }: NavigationProps) => {
...
-        currentPositionY={currentPositionY}
-        graphCardEndPosition={graphCardEndPosition}
-        currency={currency}
-        useCounterValue={useCounterValue}
-        assetPortfolio={assetPortfolio}
-        currencyBalance={currencyBalance}
-        counterValueCurrency={counterValueCurrency}
-      />
-    </TabBarSafeAreaView>
+    <ReactNavigationPerformanceView screenName={ScreenName.Asset} interactive>
```
Comment (nparigi-ledger) [inline apps/ledger-live-mobile/src/screens/WalletCentricAsset/index.tsx:301]: Nothing changed on the component tree, expect the addition of the ReactNavigationPerformanceView wrapper.

File: apps/ledger-live-mobile/src/screens/Assets/index.tsx [hunk 2]
```diff
@@ -68,62 +70,68 @@ function Assets() {
...
+        asset={item}
+        navigation={navigation}
+        sourceScreenName={ScreenName.Assets}
+      />
     ),
     [navigation],
   );
 
   return (
-    <TabBarSafeAreaView>
```
Comment (nparigi-ledger) [inline apps/ledger-live-mobile/src/screens/Assets/index.tsx:77]: Nothing changed on the component tree, expect the addition of the ReactNavigationPerformanceView wrapper.

File: apps/ledger-live-mobile/src/components/PerformanceConsole/usePerformanceReportsLog.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  return {
+    bgColor,
+    textColor,
+    statusText,
+  };
+}
+
+export default function usePerformanceReportsLog(limit = 40) {
+  // const id = useRef(0);
```
Comment (nparigi-ledger) [inline apps/ledger-live-mobile/src/components/PerformanceConsole/usePerformanceReportsLog.ts:47]: to remove

File: apps/ledger-live-mobile/src/components/PerformanceConsole/usePerformanceReportsLog.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      if (item.report.timeToBootJsMillis) {
+        setTimeToBootJs(item.report.timeToBootJsMillis);
+      }
+      setItems(currentItems => [...currentItems.slice(-(limit - 1)), item]);
+    },
+    [limit],
+  );
+  useEffect(() => {
+    const subscription = performanceReportSubject
+      // .pipe(map(item => ({ ...item, id: ++id.current })))
```
Comment (nparigi-ledger) [inline apps/ledger-live-mobile/src/components/PerformanceConsole/usePerformanceReportsLog.ts:67]: comment to remove

File: apps/ledger-live-mobile/src/screens/Accounts/AccountRow.tsx [hunk 5]
```diff
@@ -78,37 +81,43 @@ const AccountRow = ({
...
-    navigation,
-    navigationParams,
-    parentId,
-  ]);
+      if (navigationParams) {
+        startNavigationTTITimer({ source: ScreenName.Asset, uiEvent });
+        // @ts-expect-error navigagtion spread, ask your mom about it
+        navigation.navigate(...navigationParams);
+      } else if (account.type === "Account") {
+        startNavigationTTITimer({ source: ScreenName.Asset, uiEvent });
```
Comment (nparigi-ledger) [inline apps/ledger-live-mobile/src/screens/Accounts/AccountRow.tsx:94]: i think AccountRow is rendered from different places than Asset page

File: apps/ledger-live-mobile/src/components/PerformanceConsole/index.tsx [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+        position={"absolute"}
+        bottom={0}
+        alignSelf="center"
+        key={visibility}
+        entering={FadeOut.delay(2000)}
+        pb={6}
+        mb={bottom}
+        pointerEvents="none"
+      >
+        <Alert showIcon={false} title={`Analytics console: ${visibility}`} />
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/components/PerformanceConsole/index.tsx:149]: wording: "Analytics console" -> "Performance console"

**Review by ofreyssinet-ledger — COMMENTED** [review]
LGTM, just some wording to change in `PerformanceConsole/index.tsx`

**Review by ofreyssinet-ledger — APPROVED** [review]

**Pull request merged**